nsis/src/nsis/file/NsisConstants.java

```diff
@@ -6,10 +6,16 @@
 			(byte) 0xde };
 	public static final byte[] NSIS_MAGIC = { 'N', 'u', 'l', 'l', 's', 'o', 'f', 't', 'I', 'n', 's',
 			't' };
-	public static final int NB_NSIS_BLOCKS = 8;
 	public static final int NSIS_MAX_STRLEN = 1024;
 	public static final int NSIS_MAX_INST_TYPES = 32;
 	public static final byte COMPRESSION_LZMA = 0x5d;
 	public static final byte COMPRESSION_BZIP2 = 0x31;
 	public static final int COMPRESSION_LZMA_HEADER_LENGTH = 5;
+
+	// The order of the blocks is important as their ordinal value corresponds to
+	// their position in the NsisCommonHeader
+	public enum BlockHeaderType {
+		NB_PAGES, NB_SECTIONS, NB_ENTRIES, NB_STRINGS, NB_LANGTABLES, NB_CTLCOLORS, NB_BGFONT,
+		NB_DATA, NB_BLOCKS
```

Comment:
`NB_BLOCKS` here is different from the rest of entries. This pseudo-enum-entry is needed in the original C code as "last entry equals to the number of entries", thus different name, `BLOCKS_NUM`. In Java, you can use `BlockHeaderType.elements().length` to get this information.

Reply:
Updated